### ISSUE TITLE
Installing RPMs and DEBs per distribution

### ISSUE THREAD

**Issue by stebo85** [ReproNim/neurodocker#332]
Dear Neurodocker team,

I was wondering if there is a way of installing specific RPMs and DEBs in a neurodocker yaml template?
I am still trying to get minc to work nicely in neurodocker and there are prebuild package that work well:
e.g.: https://packages.bic.mni.mcgill.ca/minc-toolkit/Debian/

In the afni template I saw a DEB dependency, but it doesn't seem to support different versions of a DEB, right?

would it be ok to use gdebi/ar/rpm2cpio in the instructions set in combination with if-constructs to install the packages if required or would you favor a more general solution? 

Thank you
Steffen

**Comment by kaczmarj** [comment]
hi @stebo85 - thanks for your efforts getting minc to work nicely in neurodocker. for the deb packages, what if we use one distribution to install for all systems that use `apt`? for example, for any version, the ubuntu 18.04 deb file for that version would be installed. this _should_ work for the vast majority of installations. the general workflow would be ...

```python
# in the minc python class
url_map = {
"apt": {
  '1.9.17': 'https://packages.bic.mni.mcgill.ca/minc-toolkit/Debian/minc-toolkit-1.9.17-20190313-Ubuntu_18.04-x86_64.deb',
  '1.9.16': 'https://packages.bic.mni.mcgill.ca/minc-toolkit/Debian/minc-toolkit-1.9.16-20180117-Ubuntu_18.04-x86_64.deb',
  etc }
"yum": {...}
}

def install_minc(self):
    if self.package_manager == "apt":
        return apt_install(url_map[self.version])
    elif .... yum
```

```
# somewhere in the minc yaml template
{{ minc.install() }}
```

what do you think?

**Comment by stebo85** [comment]
Thank you @kaczmarj for the idea. I played with it a bit and I think I found a good solution: Instead of adding the download links to the python class I thought it would be cleaner to have them in the recipe. I also think we might not need the yum packages in the end, because unpacking the deb packages worked as well for a yum based distribution I tested. I hope this is useful and in line with your design philosophy?
